src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java

```diff
@@ -0,0 +1,202 @@
+
+      // create a graphics overlay and symbol to mark the extent
+      GraphicsOverlay graphicsOverlay = new GraphicsOverlay();
+      mapView.getGraphicsOverlays().add(graphicsOverlay);
+      SimpleLineSymbol boundarySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFFFF0000, 5);
+
+      // add a button to generate the geodatabase and a progress bar
+      Button generateButton = new Button("Generate Geodatabase");
+      generateButton.setDisable(true); //wait until sync task loaded
+      ProgressBar progressBar = new ProgressBar();
+      progressBar.setProgress(0.0);
+
+      // create a geodatabase sync task
+      GeodatabaseSyncTask syncTask = new GeodatabaseSyncTask(FEATURE_SERVICE_URL);
+      syncTask.loadAsync();
+      syncTask.addDoneLoadingListener(() -> generateButton.setDisable(false));
+
+      // generate the geodatabase on button click
+      generateButton.setOnMouseClicked(event -> {
+        // clear any previous operational layers and graphics if button clicked more than once
+        map.getOperationalLayers().clear();
+        graphicsOverlay.getGraphics().clear();
+
+        // show the extent used as a graphic
+        Envelope extent = mapView.getVisibleArea().getExtent();
+        Graphic boundary = new Graphic(extent, boundarySymbol);
+        graphicsOverlay.getGraphics().add(boundary);
+
+        // create generate geodatabase parameters for the current extent
+        ListenableFuture<GenerateGeodatabaseParameters> parameters = syncTask
+            .createDefaultGenerateGeodatabaseParametersAsync(extent);
+        parameters.addDoneListener(() -> {
+          try {
+            // set parameters
+            generateGeodatabaseParameters = parameters.get();
+            generateGeodatabaseParameters.setReturnAttachments(false);
+
+            // temporary file for geodatabase
+            File tempFile = File.createTempFile("gdb" + replica.getAndIncrement(), ".geodatabase");
+            tempFile.deleteOnExit();
+
+            // create and start the job
+            GenerateGeodatabaseJob job = syncTask.generateGeodatabaseAsync(generateGeodatabaseParameters, tempFile
+                .getAbsolutePath());
+            job.start();
+
+            // show progress
+            job.addProgressChangedListener(() -> {
+              int progress = job.getProgress();
+              progressBar.setProgress((double) progress / 100.0);
+            });
+
+            // get geodatabase when done
+            job.addJobDoneListener(() -> {
+              if (job.getStatus() == Job.Status.SUCCEEDED) {
+                Geodatabase geodatabase = job.getResult();
+                displayMessage("Geodatabase successfully generated", "Unregistering geodatabase since we're not " +
+                "syncing it here");
+                geodatabase.loadAsync();
+                geodatabase.addDoneLoadingListener(() ->
+                  geodatabase.getGeodatabaseFeatureTables().forEach(ft -> {
```

Comment:
would check that the geodatabase loaded properly.
`if (geodatabase.getLoadStatus() == LoadStatus.LOADED) {`